pyfstat/tcw_fstat_map_funcs.py

```diff
@@ -122,6 +124,33 @@ def get_maxF_idx(self):
         """
         return np.unravel_index(np.argmax(self.F_mn), self.F_mn.shape)
 
+    def get_logBtSG(self, windowRange):
+        """Compute (natural log of the) transient-CW Bayes-factor B_tSG = P(x|HyptS)/P(x|HypG).
+
+        Here HypG = Gaussian noise hypothesis, HyptS = transient-CW signal hypothesis.
+
+        B_tSG is marginalized over start-time and timescale of transient CW signal,
+        using given type and parameters of transient window range.
+
+        This is a python port of the `lalpulsar.ComputeTransientBstat` implementation,
+        replacing for loops by numpy operations.
+        """
+        # step through F_mn array subtract maxF and sum e^{F_mn - maxF}
+        # The maximum-likelihood Fmax is globally subtracted from F_mn, and stored separatedly in the struct, because in most
+        # expressions it is numerically more robust to compute e^(F_mn - Fmax), which at worst can underflow, while
+        # e^F_mn can overflow (for F>~700). The constant offset e^Fmax is irrelevant for posteriors (normalization constant), or
+        # can be handled separately, eg by computing log(B) = Fmax + log(sum e^(Fmn-Fmax)) for the Bayes-factor.
+        sum_eB = np.sum(np.exp(-self.maxF - self.F_mn))
+        # combine this to final log(Bstat) result with proper normalization (assuming rhohMax=1):
+        logBhat = self.maxF + np.log(sum_eB)  # unnormalized Bhat
```

Comment:
```suggestion
        logBhat = self.maxF + logsumexp(-self.maxF - self.F_mn)
```
from `scipy.special.logsumexp`. Should be more stable.

Haven't timed it though.

Reply:
That could be because logsumexp has to find the maximum internally, while in your comparison call you didn't include it, and in the Bstat use case it is already precomputed.

I can still time the real thing to make sure.

But anyway this should be part of #414 which I wanted to get merged before this one; I had just rebased this one to be able to test more easily.